File: cpp/funcaoMonotona.cpp

#include <iostream>
#include <locale.h>
#include <vector>

using namespace std;

int main(){
	setlocale(LC_ALL, "");
	
	vector<int> lista;
	int i, number;
	
	for(i=0; i<4;i++){
		cin >> number;
		lista.push_back(number);
	}

	if(lista[0] > lista[1] && lista[1] > lista[2] && lista[2] > lista[3]){
		cout << "POSSIVELMENTE ESTRITAMENTE DECRESCENTE";
	}
	else if(lista[0] < lista[1] && lista[1] < lista[2] && lista[2] < lista[3]){
		cout << "POSSIVELMENTE ESTRITAMENTE CRESCENTE";
	}
	else {cout << "FUNCAO NAO ESTRITAMENTE CRES/DECR";}

	return 0;
}

File: py/funcaoMonotona.py
nums = [int(input()) for i in range(4)]
if nums[0] < nums[1] < nums[2] < nums[3]:
    print("POSSIVELMENTE ESTRITAMENTE CRESCENTE")
elif nums[0] > nums[1] > nums[2] > nums[3]:
    print("POSSIVELMENTE ESTRITAMENTE DECRESCENTE")
else:
	print("FUNCAO NAO ESTRITAMENTE CRES/DECR")